contracts/integrations/test/function-assertions/moveStake.ts

```diff
@@ -0,0 +1,111 @@
+import {
+    GlobalStakeByStatus,
+    OwnerStakeByStatus,
+    StakeInfo,
+    StakeStatus,
+    StakingPoolById,
+    StoredBalance,
+} from '@0x/contracts-staking';
+import { constants, expect } from '@0x/contracts-test-utils';
+import { BigNumber, logUtils } from '@0x/utils';
+import { TxData } from 'ethereum-types';
+import * as _ from 'lodash';
+
+import { DeploymentManager } from '../utils/deployment_manager';
+import { FunctionAssertion } from '../utils/function_assertions';
+
+function incrementNextEpochBalance(stakeBalance: StoredBalance, amount: BigNumber): void {
+    _.update(stakeBalance, ['nextEpochBalance'], balance => (balance || constants.ZERO_AMOUNT).plus(amount));
+}
+
+function decrementNextEpochBalance(stakeBalance: StoredBalance, amount: BigNumber): void {
+    _.update(stakeBalance, ['nextEpochBalance'], balance => (balance || constants.ZERO_AMOUNT).minus(amount));
+}
+
+/**
+ * Returns a FunctionAssertion for `moveStake` which assumes valid input is provided. The
+ * FunctionAssertion checks that the staker's
+ */
+export function validMoveStakeAssertion(
+    deployment: DeploymentManager,
+    globalStake: GlobalStakeByStatus,
+    ownerStake: OwnerStakeByStatus,
+    pools: StakingPoolById,
+): FunctionAssertion<{}> {
+    const { stakingWrapper } = deployment.staking;
+
+    return new FunctionAssertion<{}>(stakingWrapper.moveStake, {
```

Comment:
Nit - These assertions are all pretty long. Splitting into helpers and/or adding comments would help w readability.